### PR TITLE
Report an error when an included file doesn't have server or client a…

### PR DESCRIPTION
**Pull request by brenthdaniel**
PR for #10762

### CODE REVIEW

**Comment by brenthdaniel** [comment]
#build

File: dev/com.ibm.ws.config/resources/com/ibm/ws/config/internal/resources/ConfigMessages.nlsprops [hunk 1]
```diff
@@ -472,5 +472,9 @@ variables.in.id.not.supported=CWWKG0104W: The ID attribute with value {1} for th
 variables.in.id.not.supported.explanation: Variable values are not allowed in ID attributes. All variables in ID attributes will be treated as literal strings.  
 variables.in.id.not.supported.useraction: Remove the variable from the value of the ID attribute.
 
+error.root.must.be.server=CWWKG0105E: The configuration document {0} must begin with the element '{1}'.
```
Comment (dmuelle) [inline dev/com.ibm.ws.config/resources/com/ibm/ws/config/internal/resources/ConfigMessages.nlsprops]: **change**
The configuration document {0} must begin with the element '{1}'.
**to**
The {0} configuration document must begin with the '{1}' element.

why is one offset with brackets and straight  quote? '{1}' Are both needed?

File: dev/com.ibm.ws.config/resources/com/ibm/ws/config/internal/resources/ConfigMessages.nlsprops [hunk 2]
```diff
@@ -472,5 +472,9 @@ variables.in.id.not.supported=CWWKG0104W: The ID attribute with value {1} for th
...
+error.root.must.be.server.explanation=Configuration must be contained in either a server element or a client element.
```
Comment (dmuelle) [inline dev/com.ibm.ws.config/resources/com/ibm/ws/config/internal/resources/ConfigMessages.nlsprops]: how will the user know whether a client or server element is required?

File: dev/com.ibm.ws.config/resources/com/ibm/ws/config/internal/resources/ConfigMessages.nlsprops (hunk 1 above)
Comment (brenthdaniel) [reply dev/com.ibm.ws.config/resources/com/ibm/ws/config/internal/resources/ConfigMessages.nlsprops]: The only reason for the quote on {1} is that without it the phrase sounds like a concept. It's better with the phrasing you suggested, but with either "server element" or "element server" it sounds like a particular server or type of server. I can remove it if needed, though.

File: dev/com.ibm.ws.config/resources/com/ibm/ws/config/internal/resources/ConfigMessages.nlsprops (hunk 2 above)
Comment (brenthdaniel) [reply dev/com.ibm.ws.config/resources/com/ibm/ws/config/internal/resources/ConfigMessages.nlsprops]: It's in the error message ( the {1} substitution )

File: dev/com.ibm.ws.config/resources/com/ibm/ws/config/internal/resources/ConfigMessages.nlsprops (hunk 1 above)
Comment (dmuelle) [reply dev/com.ibm.ws.config/resources/com/ibm/ws/config/internal/resources/ConfigMessages.nlsprops]: Ok- I see your point. The concern is that using quotes can flag an accessibility violation in some sources. I'm not sure if that would apply in this case but it's something we watch out for. I don't know where the output is expected but in a markdown source backtics (`) around `server` could do the job. [Brackets] are also a possibility.

File: dev/com.ibm.ws.config/resources/com/ibm/ws/config/internal/resources/ConfigMessages.nlsprops (hunk 2 above)
Comment (dmuelle) [reply dev/com.ibm.ws.config/resources/com/ibm/ws/config/internal/resources/ConfigMessages.nlsprops]: ok- if server or client is directly specified in the variable I think this is ok. Add "the"

The configuration must be contained in either a server element or a client element.

File: dev/com.ibm.ws.config/resources/com/ibm/ws/config/internal/resources/ConfigMessages.nlsprops (hunk 1 above)
Comment (brenthdaniel) [reply dev/com.ibm.ws.config/resources/com/ibm/ws/config/internal/resources/ConfigMessages.nlsprops]: We use brackets elsewhere for config messages, so I'll go with that.

File: dev/com.ibm.ws.config/resources/com/ibm/ws/config/internal/resources/ConfigMessages.nlsprops [hunk 3]
```diff
@@ -472,5 +472,9 @@ variables.in.id.not.supported=CWWKG0104W: The ID attribute with value {1} for th
 variables.in.id.not.supported.explanation: Variable values are not allowed in ID attributes. All variables in ID attributes will be treated as literal strings.  
 variables.in.id.not.supported.useraction: Remove the variable from the value of the ID attribute.
 
+error.root.must.be.server=CWWKG0105E: The {0} configuration document must begin with the [{1}] element.
+error.root.must.be.server.explanation=The configuration must be contained in either a server element or a client element.
```
Comment (dmuelle) [inline dev/com.ibm.ws.config/resources/com/ibm/ws/config/internal/resources/ConfigMessages.nlsprops]: should this message have a colon (:) like the other two or is it supposed to have the (=) ?

Comment (brenthdaniel) [reply dev/com.ibm.ws.config/resources/com/ibm/ws/config/internal/resources/ConfigMessages.nlsprops]: I'm surprised the build didn't fail for that.. but it looks like the previous message ID in the file has the same issue. I fixed both of them.

**Comment by brenthdaniel** [comment]
#build

**Comment by brenthdaniel** [comment]
#run-libby-bot

**Pull request merged**